airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/StateDecoratingIterator.java

```diff
@@ -59,24 +69,36 @@ protected AirbyteMessage computeNext() {
         }
       }
 
-      return message;
-    } else if (!hasEmittedState) {
-      final AirbyteStateMessage stateMessage = stateManager.updateAndEmit(pair, maxCursor);
-      LOGGER.info("State Report: stream name: {}, original cursor field: {}, original cursor {}, cursor field: {}, new cursor: {}",
-          pair,
-          stateManager.getOriginalCursorField(pair).orElse(null),
-          stateManager.getOriginalCursor(pair).orElse(null),
-          stateManager.getCursorField(pair).orElse(null),
-          stateManager.getCursor(pair).orElse(null));
-      if (stateManager.getCursor(pair).isEmpty()) {
-        LOGGER.warn("Cursor was for stream {} was null. This stream will replicate all records on the next run", pair);
+      if (stateEmissionFrequency > 0 && recordCount % stateEmissionFrequency == 0) {
+        final boolean isFinalState = !messageIterator.hasNext();
+        emitStateMessage(isFinalState);
```

Comment:
We don't have to. The purpose is to prevent sending the same state message twice if this intermediate message happens to be the last one. Given that it is really simple, I think it's better to do it. In this way, we don't need to worry about this edge case in unit testing.